packages/tpetra/core/ext/TpetraExt_MatrixMatrix_def.hpp

```diff
@@ -622,11 +551,16 @@ add (const Scalar& alpha,
 {
   typedef Map<LocalOrdinal,GlobalOrdinal,Node>                     map_type;
   typedef CrsMatrix<Scalar,LocalOrdinal,GlobalOrdinal,Node>  crs_matrix_type;
+  //Check that A,B are fillComplete before getting B's column map
+  TEUCHOS_TEST_FOR_EXCEPTION
+    (! A.isFillComplete () || ! B.isFillComplete (), std::invalid_argument,
+     "TpetraExt::MatrixMatrix::add(): A and B must both be fill complete.");
   Teuchos::RCP<const map_type> CrowMap = transposeB ?  B.getDomainMap() : B.getRowMap();
```

Comment:
It's called the "row Map," and it's used as a row Map below, but it's either a domain or range Map.  Is that OK?

Reply:
Good catch! Neither B's domain map nor col map are correct for C row map (if transposeB). I ended up copying the explicit transpose of B into this function, since the RowMapTransposer gives Bprime the correct rowmap.